### PR TITLE
AREA-5 - Setup architecture

### PR DESCRIPTION
**Pull request by Yann-Masson**
The project architecture is now completely setup (Well, it should be...).
Because it's our first time with flutter, It is almost certain that this architecture is not definitive and will evolve in the future.

As you'll see in this PR, a lot of things are from the flutter example app. We'll customize and clean that later. We have to start some other thins soon as possible so that's why I haven't clean up this shit.

### CODE REVIEW

File: mobile/pubspec.yaml
Comment (Thomaltarix) [inline mobile/pubspec.yaml:1]: You can add the bloc dependencies so that we don't need to modify it later on

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
Since you're doing architecture, it would be nice not to have to modify it later.

**Review by TekMath — CHANGES_REQUESTED** [review]
Please update your structure with real MVVM (view layers)

**Comment by Yann-Masson** [comment]
> Since you're doing architecture, it would be nice not to have to modify it later.

As explain in my PR description, we're totally new to Flutter. So this architecture might be update in time.

**Comment by Yann-Masson** [comment]
> Please update your structure with real MVVM (view layers)

We don't use the _real_ MVVM project structure. We use a custom architecture that suits us best.

**Review by Thomaltarix — APPROVED** [review]

**Review by TekMath — APPROVED** [review]

**Pull request merged**